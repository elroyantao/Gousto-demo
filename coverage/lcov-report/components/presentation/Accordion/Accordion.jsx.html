<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/presentation/Accordion/Accordion.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">components/presentation/Accordion</a> Accordion.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">6.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">6.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/15</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react'
import PropTypes from 'prop-types'
&nbsp;
export default class Accordion extends Component {
<span class="fstat-no" title="function not covered" >  co</span>nstructor(props) {
<span class="cstat-no" title="statement not covered" >    super(props)</span>
<span class="cstat-no" title="statement not covered" >    this.state = {</span>
      expandedHeight: 0,
      expanded: false
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentDidMount() {
<span class="cstat-no" title="statement not covered" >    this.updateExpandedHeight()</span>
  }
&nbsp;
  getContentHeight = <span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >this.accordionWrapper.scrollHeight</span>
&nbsp;
  handleToggle = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      expanded: !this.state.expanded
    })
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  up</span>dateExpandedHeight() {
    const newHeight = <span class="cstat-no" title="statement not covered" >this.getContentHeight()</span>
<span class="cstat-no" title="statement not covered" >    if (newHeight === undefined) {</span>
<span class="cstat-no" title="statement not covered" >      return</span>
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    if (newHeight !== this.state.expandedHeight) {</span>
<span class="cstat-no" title="statement not covered" >      this.setState({ expandedHeight: newHeight })</span>
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  re</span>nder() {
    const { header, children } = <span class="cstat-no" title="statement not covered" >this.props</span>
    const { expandedHeight, expanded } = <span class="cstat-no" title="statement not covered" >this.state</span>
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;div className="Accordion"&gt;
        &lt;div
          className="Accordion-header"
          onClick={this.handleToggle}
          style={{ cursor: children ? 'pointer' : 'auto' }}
        &gt;
          { header }
        &lt;/div&gt;
        &lt;div
          className="Accordion-wrapper"
          ref={<span class="fstat-no" title="function not covered" >(e</span>lement) =&gt; { <span class="cstat-no" title="statement not covered" >this.accordionWrapper = element </span>}}
          style={{ maxHeight: expanded ? expandedHeight : 0, overflow: 'hidden' }}
        &gt;
          { children &amp;&amp;
            &lt;div className="Accordion-content"&gt;
              {children}
            &lt;/div&gt;
          }
        &lt;/div&gt;
&nbsp;
      &lt;/div&gt;
    )
  }
}
&nbsp;
Accordion.propTypes = {
  children: PropTypes.node,
  header: PropTypes.string
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Jul 20 2017 11:00:38 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
