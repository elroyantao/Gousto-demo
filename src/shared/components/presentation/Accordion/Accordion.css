:root {
  --pagePadding: 10px;
}

.Accordion {
  margin: 5px 0;

  color: gray(85);
}

.Accordion-header {
  cursor: pointer;
  border-top: solid gray(80) 1px;
  border-bottom: solid gray(80) 1px;
  font-weight: bold;
  padding: 5px var(--pagePadding);
}

.Accordion-header--open {
  composes: Accordion-header;
}

.Accordion-wrapper {
  overflow: hidden;
  font-style: italic;
}

.Accordion-content {
  padding: 5px var(--pagePadding);
}
