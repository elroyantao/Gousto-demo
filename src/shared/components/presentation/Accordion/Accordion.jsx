import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class Accordion extends Component {
  static propTypes = {
    header: PropTypes.string,
    children: PropTypes.node
  }

  constructor(props) {
    super(props)
    this.state = {
      expandedHeight: 0,
      expanded: false
    }
  }

  componentDidMount() {
    this.updateExpandedHeight()
  }

  getContentHeight = () => this.accordionWrapper.scrollHeight

  handleToggle = () => {
    this.setState({
      expanded: !this.state.expanded
    })
  }

  updateExpandedHeight() {
    const newHeight = this.getContentHeight()
    if (newHeight === undefined) {
      return
    }

    if (newHeight !== this.state.expandedHeight) {
      this.setState({ expandedHeight: newHeight })
    }
  }

  render() {
    const { header, children } = this.props
    const { expandedHeight, expanded } = this.state
    return (
      <div className="Accordion">
        <div
          className="Accordion-header"
          onClick={this.handleToggle}
          style={{ cursor: children ? 'pointer' : 'auto' }}
        >
          { header }
        </div>
        <div
          className="Accordion-wrapper"
          ref={(element) => { this.accordionWrapper = element }}
          style={{ maxHeight: expanded ? expandedHeight : 0, overflow: 'hidden' }}
        >
          { children &&
            <div className="Accordion-content">
              {children}
            </div>
          }
        </div>

      </div>
    )
  }
}
