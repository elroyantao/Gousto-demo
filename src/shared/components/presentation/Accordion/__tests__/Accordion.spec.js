import React from 'react'

import testComponentHelper from '../../../../lib/test-helper'
import Accordion from '../Accordion'

describe('<Accordion />', () => {
  const initialProps = {
    header: 'this is a header'
  }
  const renderComponent = testComponentHelper(Accordion)

  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialProps).getTree())
        .toMatchSnapshot()
    })
    it('with child node', () => {
      expect(renderComponent({
        ...initialProps,
        children: <div>this is a body</div>
      }).getTree())
        .toMatchSnapshot()
    })
    it('when extended height is set and is expanded', () => {
      const { wrapper, getTree } = renderComponent({
        ...initialProps,
        children: <div>this is a body</div>
      })
      wrapper.setState({
        expanded: true,
        expandedHeight: 100
      })
      expect(getTree()).toMatchSnapshot()
    })
    it('when extended height is set and is not expanded', () => {
      const { wrapper, getTree } = renderComponent({
        ...initialProps,
        children: <div>this is a body</div>
      })
      wrapper.setState({
        expanded: false,
        expandedHeight: 100
      })
      expect(getTree()).toMatchSnapshot()
    })
  })
  describe('@lifecycle', () => {
    describe('componentDidMount', () => {
      let initialScrollHeight
      let renderedComponent
      beforeEach(() => {
        renderedComponent = renderComponent({
          ...initialProps
        })
        initialScrollHeight = renderedComponent.instance.state.expandedHeight
      })

      it('do nothing if scrollHeight is undefined', () => {
        const { instance, wrapper } = renderedComponent
        wrapper.find('.Accordion-wrapper').node.ref({ scrollHeight: undefined })
        expect(instance.state.expandedHeight).toBe(initialScrollHeight)
        instance.componentDidMount()
        expect(instance.state.expandedHeight).toBe(initialScrollHeight)
      })

      it('do nothing if scrollHeight is remains same', () => {
        const { instance, wrapper } = renderedComponent
        wrapper.find('.Accordion-wrapper').node.ref({ scrollHeight: initialScrollHeight })
        expect(instance.state.expandedHeight).toBe(initialScrollHeight)
        instance.componentDidMount()
        expect(instance.state.expandedHeight).toBe(initialScrollHeight)
      })

      it('update expandedHeight if changed', () => {
        const { instance, wrapper } = renderedComponent
        expect(instance.state.expandedHeight).toBe(initialScrollHeight)
        wrapper.find('.Accordion-wrapper').node.ref({ scrollHeight: 50 })
        instance.componentDidMount()
        expect(instance.state.expandedHeight).toBe(50)
      })
    })
  })

  describe('@events', () => {
    describe('onClick header', () => {
      it('should toggle the state.expanded', () => {
        const { wrapper, instance } = renderComponent(initialProps)
        const initialExpanded = instance.state.expanded
        wrapper.find('.Accordion-header').simulate('click')
        expect(instance.state.expanded).toBe(!initialExpanded)
      })
    })
  })
})
