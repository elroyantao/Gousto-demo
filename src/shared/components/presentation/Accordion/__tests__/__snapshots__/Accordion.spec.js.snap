// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> @renders in default state 1`] = `
<div
  className="Accordion"
>
  <div
    className="Accordion-header"
    onClick={[Function]}
    style={
      Object {
        "cursor": "auto",
      }
    }
  >
    this is a header
  </div>
  <div
    className="Accordion-wrapper"
    style={
      Object {
        "maxHeight": 0,
        "overflow": "hidden",
      }
    }
  />
</div>
`;

exports[`<Accordion /> @renders when extended height is set and is expanded 1`] = `
<div
  className="Accordion"
>
  <div
    className="Accordion-header"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    this is a header
  </div>
  <div
    className="Accordion-wrapper"
    style={
      Object {
        "maxHeight": 100,
        "overflow": "hidden",
      }
    }
  >
    <div
      className="Accordion-content"
    >
      <div>
        this is a body
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> @renders when extended height is set and is not expanded 1`] = `
<div
  className="Accordion"
>
  <div
    className="Accordion-header"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    this is a header
  </div>
  <div
    className="Accordion-wrapper"
    style={
      Object {
        "maxHeight": 0,
        "overflow": "hidden",
      }
    }
  >
    <div
      className="Accordion-content"
    >
      <div>
        this is a body
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> @renders with child node 1`] = `
<div
  className="Accordion"
>
  <div
    className="Accordion-header"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    this is a header
  </div>
  <div
    className="Accordion-wrapper"
    style={
      Object {
        "maxHeight": 0,
        "overflow": "hidden",
      }
    }
  >
    <div
      className="Accordion-content"
    >
      <div>
        this is a body
      </div>
    </div>
  </div>
</div>
`;
